### DISCUSSION TITLE
Flow control features

### DISCUSSION THREAD

**Discussion by lautit** [Nain57/Smart-AutoClicker#335] [General Feature Requests]
### Event trigger once

Currently, this pattern can be implemented using the "Change event state" action, with the "New event state" set to "Disable" and selecting the same Event as target.

To improve this pattern, my suggestions would be:
- an Action "Disable self"
- a toggle in the Event Config "Trigger event once" or "Disable after trigger"

### Multiple event toggling

The "Change event state" action allows to control the flow of image recognition, efectivelly detecting true/false scenarios using same or multiple Condition images. In order to do so, there are cases where it is needed to "Enable" some Events and "Disable" others.

To improve this pattern, my suggestion is a "Change multiple event states" action, simplifying the UI and reducing the visual complexity.

### Bifurcated actions

As stated in the previous point, there is a pattern which enables to do certain actions in case of Condition Visibility. This forces the creation of multiple Events, with the same Condition.

To improve this pattern, my suggestion is a toggle in the Event Config "Bifurcated actions", which lets you define a pair of Actions for the simplest "Present" or "Absent" Condition Visibilities.

**Comment by lautit** [comment]
I've packed all of these features together, as they seem to be very much related between each other, specifically with the "Change event state" action. Am I being logical here or would it be better to split them up?

**Comment by Nain57** [comment]
Closed as implemented. (a long ago)
There is now multiple flow control feature, check the wiki: https://github.com/Nain57/Smart-AutoClicker/wiki
